src/Examine.Test/Examine.Lucene/Search/FluentApiTests.cs

```diff
@@ -86,21 +110,87 @@ public void NativeQuery_Single_Word()
+                    Assert.AreEqual(1, facetResults.Last().Value);
+                }
+                else
+                {
+                    var results = query.Execute();
+
+                    Assert.AreEqual(2, results.TotalItemCount);
+                }
+            }
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void Uppercase_Category(bool withFacets)
+        {
+            var fieldDefinitionCollection = withFacets ?
+                new FieldDefinitionCollection(new FieldDefinition("parentID", FieldDefinitionTypes.Integer), new FieldDefinition("nodeName", FieldDefinitionTypes.FacetFullText))
+                : new FieldDefinitionCollection(new FieldDefinition("parentID", FieldDefinitionTypes.Integer));
+            var analyzer = new StandardAnalyzer(LuceneInfo.CurrentVersion);
+            using (var luceneDir = new RandomIdRAMDirectory())
+            using (var indexer = GetTestIndex(
+                luceneDir,
+                analyzer,
+                fieldDefinitionCollection))
+            {
+                indexer.IndexItems(new[] {
+                    ValueSet.FromObject(1.ToString(), "cOntent",
+                        new { nodeName = "location 1", bodyText = "Zanzibar is in Africa"}),
+                    ValueSet.FromObject(2.ToString(), "cOntent",
+                        new { nodeName = "location 2", bodyText = "In Canada there is a town called Sydney in Nova Scotia"}),
+                    ValueSet.FromObject(3.ToString(), "cOntent",
+                        new { nodeName = "location 3", bodyText = "Sydney is the capital of NSW in Australia"})
+                    });
+
+                var searcher = indexer.Searcher;
+
+                var query = searcher.CreateQuery("cOntent").All();
+
+                Console.WriteLine(query);
+
+                if (withFacets)
+                {
+                    var results = query.WithFacets(facets => facets.Facet("nodeName")).Execute();
+
+                    var facetResults = results.GetFacet("nodeName");
+
+                    Assert.AreEqual(3, results.TotalItemCount);
+                    Assert.AreEqual(3, facetResults.Count());
+                }
+                else
+                {
+                    var results = query.Execute();
 
-                Assert.AreEqual(2, results.TotalItemCount);
+                    Assert.AreEqual(3, results.TotalItemCount);
+                }
             }
         }
 
         [Test]
-        public void Uppercase_Category()
+        public void FacetsConfig_SetIndexName_FullText()
         {
+            var fieldDefinitionCollection = new FieldDefinitionCollection(new FieldDefinition("parentID", FieldDefinitionTypes.Integer), new FieldDefinition("nodeName", FieldDefinitionTypes.FacetFullText));
+
+            var facetsConfig = new FacetsConfig();
+            facetsConfig.SetIndexFieldName("nodeName", "facet_nodeName");
```

Comment:
This test scenario should not occur. SetIndexFieldName is only meant to be called when there is a separate Taxonomy index in use. See the Lucene.NET docs for this method.

Reply:
I know this is the case according to the doc. But this works for these use cases as well. So therefore it makes sense to test.

Reply:
While it does work per say, its not possible to then identify taxonomy vs non taxonomy indexed facets.

Reply:
So you suggest removing support for using the indexFieldName just because there's no other way to identify if a field is using taxonomy? Seems like a werid approach when Lucene support the facet field name as a parameter in the facet counters. This is also the only way I know that you can use a custom field name for storing facets which might be a useful feature for some.

Reply:
Good point. I'll have a think about another way.